curriculum/challenges/english/07-scientific-computing-with-python/learn-interfaces-by-building-an-equation-solver/663a32735b317af9812eb0d7.md

```diff
@@ -0,0 +1,117 @@
+---
+id: 663a32735b317af9812eb0d7
+title: Step 30
+challengeType: 20
+dashedName: step-30
+---
+
+# --description--
+
+In linear equations in the form \\( ax + b = 0 \\), the slope is simply the coefficient \\( a \\), and the y-intercept is the coefficient \\( b \\).
+
+You are going to use the `analyze` method to provide additional information about the equation. Inside the `analyze` method, append a string in the form `f'slope = {}\ny-intercept = {}'` to the `details` attribute: fill the curly braces so that they evaluate to the slope and the y-intercept, respectively.
```

Comment:
It's expected, since we are printing the list. If we print the item it shows in separate lines. Maybe I can ask to print the string instead of the list.